### DISCUSSION TITLE
AppData

### DISCUSSION THREAD

**Discussion by TortoiseWolfe** [wixtoolset/issues#7797] [Questions]
I'm developing a plugin for Revit, it has a dedicated subfolder n `Appdata` so it already has several other files that I shouldn't touch, but when I try to build it says the user profile is not listed in the RemoveFile table.  But it shouldn't be trying to delete any of those folders, the other plugins will still be using them.

```
<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs">
	<Package Name="acme" Manufacturer="TODO Manufacturer" Version="1.0.0.0" UpgradeCode="db4bd8c4-7c09-46d7-8fb0-457d2ae2d0ca">
		<MajorUpgrade DowngradeErrorMessage="!(loc.DowngradeError)" />
		<Feature Id="MainFeature">
			<ComponentRef Id="Component1" />
			<ComponentRef Id="Component2" />
		</Feature>

		<Directory Id="TARGETDIR" Name="SourceDir">
			<Directory Id="AppDataFolder">
				<Directory Id="AutodeskFolder" Name="Autodesk">
					<Directory Id="RevitFolder" Name="Revit">
						<Directory Id="AddinsFolder" Name="Addins">
							<Directory Id="VersionFolder" Name="2023">
								<Component Id="Component1" Guid="82FC8015-5992-4B0E-9E40-6BE6E9CF883D">
									<RegistryValue Root="HKCU" Key="Software\[Manufacturer]\[ProductName]" Name="installed" Type="integer" Value="1" KeyPath="yes"/>
									<File Id="File1" Source="C:\Users\Player One\source\repos\TortoiseWolfe\T23\T23.addin" />
									<RemoveFile Id="RemoveFile1" Name="T23.addin" On="uninstall"/>
								</Component>
								<Component Id="Component2" Guid="0D345F7F-A743-4B1E-898D-68ED56437699">
									<RegistryValue Root="HKCU" Key="Software\[Manufacturer]\[ProductName]" Name="installed" Type="integer" Value="1" KeyPath="yes"/>
									<File Id="File2" Source="C:\Users\Player One\source\repos\TortoiseWolfe\T23\bin\Debug\T23.dll" />
									<RemoveFile Id="RemoveFile2" Name="T23.dll" On="uninstall"/>
								</Component>
							</Directory>
						</Directory>
					</Directory>
				</Directory>
			</Directory>
		</Directory>
	</Package>
</Wix>

```

![image](https://github.com/wixtoolset/issues/assets/101219421/14798d58-8fb4-42aa-a7fc-22c37692da33)

**Comment by robmen** [comment]
I think this is a case where the Windows Installer's static analysis (ICEs) are leading you down the wrong path. The Windows Installer has some very particular ways of viewing the user profile, and they come from a time before applications started installing software there (aka: before UAC).

I [just mentioned in another discussion](https://github.com/orgs/wixtoolset/discussions/7801) how I happen to go off on a tangent and talk about the Windows Installer per-user design decisions in a good bit of detail in the latter part of [Episode 51 in the Deployment Dojo](https://www.youtube.com/live/pOIdji4NLVU?si=VlaVOcGtl26gf5pm).

In this case, I think the ICEs are wrong (I don't say that very often), and you should remove everything but the Files from your Components. The RegistryValue and RemoveFile aren't going to help your adding actually roam your add-in.

PS: It is really strange that Autodesk would choose to place the add-ins folder in the user's roaming profile. I bet IT admins dislike that.
